sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -226,3 +221,30 @@ export class AzurePowerShellCredential implements TokenCredential {
     });
   }
 }
+
+/**
+ *
+ * @internal
+ */
+export async function parseJsonToken(
+  result: string,
+): Promise<{ Token: string; ExpiresOn: string }> {
+  const jsonRegex = /{[^{}]*}/g;
+  const matches = result.match(jsonRegex);
+  let resultWithoutToken = result;
+  if (matches) {
+    try {
+      for (const item of matches) {
```

Comment:
Hmm if the first match is malformed, we'll not consider later ones, is that right?

Reply:
No no we will consider later ones. I have an example in tests.

Reply:
I had added an inner tr-catch too with the catch just being continue

Reply:
added back